rlclientlib/model_mgmt/model_downloader.cc

```diff
@@ -30,11 +30,6 @@ namespace reinforcement_learning { namespace model_management {
     model_data md;
     RETURN_IF_FAIL(_ptrans->get_data(md, status));
 
-    // If the data size is zero, it's not a valid model
```

Comment:
Today when the loop is created a zero length model file is also created.  We test for the existence of this model file during init as an additional guard to make sure that the loop is created properly.  However, we don't hydrate a vw model with this empty file.  Will that continue to work?

Reply:
Yes - what happens now is that the i_model implementation is expected to specify whether it is ready to handle inference at update() time. Thus, models that are self-contained can participate in this without the model-downloader having special-case logic.

I think there is more refactoring to be done here, but that would likely require a change in how configuration works.